File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
export class Squares {
  #squareOfSum;
  #sumOfSquares;
 
  constructor(n) {
    this.#squareOfSum = 0;
    this.#sumOfSquares = 0;
    for (let i = 1; i <= n; ++i) {
      this.#sumOfSquares += (i ** 2);
      this.#squareOfSum += i;
    } 
    this.#squareOfSum *= this.#squareOfSum;
  }

  get sumOfSquares() {
    return this.#sumOfSquares;
  }

  get squareOfSum() {
    return this.#squareOfSum;
  }

  get difference() {
    return this.#squareOfSum - this.#sumOfSquares;
  }
}

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {
    auto square_of_sum(std::size_t number) -> uint64_t {
        const auto sum_of_numbers = (number * (number + 1)) / 2;
        return sum_of_numbers * sum_of_numbers;
    }
    
    auto sum_of_squares(std::size_t number) -> uint64_t {
        return (number * (number + 1) * ((2 * number) + 1)) / 6;        
    }

    auto difference(std::size_t number) -> uint64_t {
        return square_of_sum(number) - sum_of_squares(number);
    }
}  // namespace difference_of_squares
